File: backend/FitnessApp/src/main/java/com/andrea360/fitnessapp/service/payment/PaymentService.java
package com.andrea360.fitnessapp.service.payment;

import com.andrea360.fitnessapp.model.Purchase;
import com.andrea360.fitnessapp.dto.payment.ConfirmPaymentRequest;
import com.andrea360.fitnessapp.dto.payment.CreatePaymentIntentRequest;
import com.andrea360.fitnessapp.dto.payment.CreatePaymentIntentResponse;

public interface PaymentService {
    CreatePaymentIntentResponse createPaymentIntent(String email, CreatePaymentIntentRequest request);
    Purchase confirmPayment(String email, ConfirmPaymentRequest request);
}

File: frontend/src/services/paymentService.ts
import axiosClient from '../axios/axiosClient';
import type { CreatePaymentIntentRequest, CreatePaymentIntentResponse, ConfirmPaymentRequest } from '../types/Payment';
import type { Purchase } from '../types/Purchase';

export const createPaymentIntent = async (data: CreatePaymentIntentRequest): Promise<CreatePaymentIntentResponse> => {
  const response = await axiosClient.post<CreatePaymentIntentResponse>('/payments/create-intent', data);
  return response.data;
};

export const confirmPayment = async (data: ConfirmPaymentRequest): Promise<Purchase> => {
  const response = await axiosClient.post<Purchase>('/payments/confirm', data);
  return response.data;
};